### PR TITLE
Shuya currservice

### PR DESCRIPTION
**Pull request by shuyashou**
# Overview

In this PR, I change the way of URL construction according to what Bryan said from last time. Instead of string#format I use urivaribles this time. Since Kyrie push his code of UCSBCurriculumController to my branch (which isn't recommend will never do again), this PR contains both service and controller. That's why I address 2 issues here.

# Issues Addressed
#4 
#5 

# Details

### CODE REVIEW

**Review by WAKACHUU — DISMISSED** [review]
LGTM

**Review by binyuzhong — DISMISSED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package edu.ucsb.cs156.example.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+//-no longer used library-//
+// import com.fasterxml.jackson.databind.ObjectMapper;
+// import org.slf4j.Logger;
+// import org.slf4j.LoggerFactory;
+//------------------------//
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java]: Why are these lines here?  They should be deleted.

File: src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import edu.ucsb.cs156.example.services.UCSBCurriculumService;
+
+@RestController
+@RequestMapping("/api/public")
+public class UCSBCurriculumController {
+    // private final Logger logger = LoggerFactory.getLogger(BasicSearchController.class);
+    // private ObjectMapper mapper = new ObjectMapper();
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java]: Why are these commented out lines here?
Please delete

File: src/main/java/edu/ucsb/cs156/example/services/UCSBCurriculumService.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+ */
+@Service
+public class UCSBCurriculumService  {
+
+    private Logger logger = LoggerFactory.getLogger(UCSBCurriculumService.class);
+
+    @Value("${app.ucsb.api.consumer_key}")
+    private String apiKey;
+
+    //private RestTemplate restTemplate = new RestTemplate();
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/example/services/UCSBCurriculumService.java]: Please delete commented out lines

File: src/main/java/edu/ucsb/cs156/example/services/UCSBCurriculumService.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+        Map<String,String> uriVariables = Map.of("quarter_in",quarter,"subjectCode_in",subjectArea,"objLevelCode_in",courseLevel);
+
+        if(courseLevel.equals("A")){
+            URL = "https://api.ucsb.edu/academics/curriculums/v1/classes/search?quarter={quarter_in}&subjectCode={subjectCode_in}&pageNumber=1&pageSize=100&includeClassSections=true";
+
+            uriVariables = Map.of("quarter_in",quarter,"subjectCode_in",subjectArea);
+        }
+
+        //logger.info("URL=" + URL);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/example/services/UCSBCurriculumService.java]: Please delete commented out lines.

File: src/test/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumControllerTests.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package edu.ucsb.cs156.example.controllers;
+
+import edu.ucsb.cs156.example.config.SecurityConfig;
+import edu.ucsb.cs156.example.repositories.UserRepository;
+// import edu.ucsb.cs156.example.documents.Course;
+// import edu.ucsb.cs156.example.documents.CoursePage;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumControllerTests.java]: delete

File: src/test/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumControllerTests.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+// import java.io.PrintWriter;
+// import java.util.ArrayList;
+
+// import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumControllerTests.java]: delete

File: src/test/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumControllerTests.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+// @Import(SecurityConfig.class) applies the security rules 
+// so that /api/public/** endpoints don't require authentication.
+// Otherwise you may get authorization errors when running the test
+
+@WebMvcTest(value = UCSBCurriculumController.class)
+@Import(SecurityConfig.class)
+public class UCSBCurriculumControllerTests {
+
+    // private final Logger logger = LoggerFactory.getLogger(UCSBCurriculumControllerTests.class);
+    // private ObjectMapper mapper = new ObjectMapper();
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumControllerTests.java]: delete

File: src/test/java/edu/ucsb/cs156/example/services/UCSBCurriculumServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+    @Test
+    public void test_getJSON_success() throws Exception {
+/*
+
+        level = "A";
+        result = ucs.getJSON(subjectArea, quarter, level);
+
+        assertEquals(expectedResult, result);
+*/
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/example/services/UCSBCurriculumServiceTest.java]: delete

**Review by pconrad — CHANGES_REQUESTED** [review]
Please delete commented out code.

File: src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import edu.ucsb.cs156.example.services.UCSBCurriculumService;
+
+@RestController
+@RequestMapping("/api/public")
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java]: I don't think this is a good base URL for this controller. This is a UCSB Curriculum controller, not a public controller. Maybe a base URL like `/api/curriculum` would be better?

File: src/main/java/edu/ucsb/cs156/example/services/UCSBCurriculumService.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
+package edu.ucsb.cs156.example.services;
+import java.util.Map;
+import java.util.Arrays;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.web.client.RestTemplateBuilder;
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/cs156/example/services/UCSBCurriculumService.java]: nitpick: can we keep all the `org.springframework` imports together and alphabetized?

**Review by andrewhlu — CHANGES_REQUESTED** [review]
Mostly looks good, but I have one suggestion (and one small nitpick), see below.

File: src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java (hunk 9 above)
Comment (pconrad) [reply src/main/java/edu/ucsb/cs156/example/controllers/UCSBCurriculumController.java]: Agreed.

**Review by pconrad — CHANGES_REQUESTED** [review]
Agreed.  And the best time to fix that is before lots of other code depends on it.

**Review by noahzlutz — APPROVED** [review]
LGTM changes request were made

**Review by ericxu002 — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM
